keras_nlp/src/models/mix_transformer/mix_transformer_backbone_test.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2024 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import pytest
+from keras import models
+
+from keras_nlp.src.models.mix_transformer.mix_transformer_backbone import (
+    MiTBackbone,
+)
+from keras_nlp.src.tests.test_case import TestCase
+
+
+class MiTBackboneTest(TestCase):
+    def setUp(self):
+        self.init_kwargs = {
```

Comment:
once we have things pulled into the config, the goal with our test model should be to use very few parameters and layers so things test fast. E.g. test with an image size `(16, 16, 3)` and max hidden state size of 32 or something like that

Reply:
64 is the minimum image size it is required I think, else it would fail like https://colab.sandbox.google.com/gist/sachinprasadhs/09e6c824daa2d6a3093f9ca44673d9b2/backbine-test.ipynb

Reply:
Why would it fail? I don't see the failure. Ideally once you pull more of these magic numbers into config, things could work smaller, but probably missing something.

Reply:
I think it's because of the hierarchical encoder architecture  according to the paper here, in the encoder part, the transformer block 4 (P4) is having H/32 & W/32
![image](https://github.com/user-attachments/assets/903c057a-2918-4878-b741-16e8d168cb9a)

Reply:
Can we just test with two layers instead of four? Not that it's a big deal, but it would be a good test that our config "works" and can create different shaped models than the default.

Reply:
Done, reduced the image size to 16, 16 and restricted number of layers to 2.